internal/datastore/data_store_eval_impl.go

```diff
@@ -9,13 +9,13 @@ import (
 )
 
 type dataStoreEvaluatorDataProviderImpl struct {
```

Comment:
This change is because the existing read/write enabled `DataStore` interface isn't actually needed in the evaluator. It only needs to be able to retrieve segments/flags.